tempodb/backend/raw.go

```diff
@@ -40,6 +40,8 @@ type RawWriter interface {
 type RawReader interface {
 	// List returns all objects one level beneath the provided keypath
 	List(ctx context.Context, keypath KeyPath) ([]string, error)
+	// ListBlocks returns the IDs for blocks and compacted blocks at the keypath
```

Comment:
I took this approach because I wanted to avoid touching the existing implementations.  `List()` is a pretty generic way to get the contents of a path, but what we're aiming at here is a special method of getting the blocks we know about, which feels like it deserves its own method.  We could extend `List()` perhaps to include an additional argument, but then this would enable the specific features, and so it feels like we're back to the `RawReader` having special handling for special cases.

Since we're dealing with the specifics around implementation, I'm not sure what that would look like, but I'm all ears if you have thoughts on how to unify them.

Reply:
this is harder. let's sync offline. i'm against adding any "tempo" specific information to this layer as it's mostly tempo agnostic now, but this may be an exception if there really is substantial gains by splitting this off.

Reply:
I've refactored to include a `Find()` method that allows options to be passed in.  Future changes can be made to the opts struct to allow more complex searching, but for now I think this works and keeps the code agnostic as to what is being searched from the backend.